### ISSUE TITLE
[k8s][spot] Support spot controller on Kubernetes or local clusters

### ISSUE THREAD

**Issue by romilbhardwaj** [skypilot-org/skypilot#2249]
Kubernetes support in #2096 cannot launch spot controller because the spot controller requires autostop capability:

```
Creating a new cluster: "sky-spot-controller-e53e0cdf" [1x Kubernetes(2CPU--2GB, disk_size=50)].
I 07-13 20:36:41 cloud_vm_ray_backend.py:3945] Tip: to reuse an existing cluster, specify --cluster (-c). Run `sky status` to see existing clusters.
W 07-13 20:36:43 cloud_vm_ray_backend.py:2000] sky.exceptions.NotSupportedError: The following features are not supported by Kubernetes:
W 07-13 20:36:43 cloud_vm_ray_backend.py:2000]  Feature   Reason
W 07-13 20:36:43 cloud_vm_ray_backend.py:2000]  autostop  Kubernetes does not support stopping VMs.
```

A temporary workaround is to set `SPOT_CONTROLLER_IDLE_MINUTES_TO_AUTOSTOP = None` in [spot/constants.py](https://github.com/skypilot-org/skypilot/blob/master/sky/spot/constants.py#L3).

A cleaner solution would be a toggle the autostop for spot controller based on the cloud it is being provisioned on.

**Comment by concretevitamin** [comment]
Bump.

At least three users have said that a controller VM on the cloud for them, despite using autostop, is not entirely ideal due to costs. They have requested the feature to run spot controllers on a local machine / existing VM.

**Comment by Michaelvll** [comment]
This should be fixed by #3109

**Comment by Michaelvll** [comment]
This has been fixed by #3377
